code-challenges.test.js

```diff
@@ -14,45 +14,134 @@
+    // Expected output: "3cc3ntr1c"
+    
+  
+    it("takes in a string of letters and converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+      expect(letsMakeItUnreadable(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+      expect(letsMakeItUnreadable(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+      expect(letsMakeItUnreadable(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+  })
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
 
+// INPUT: a string of letters
+// OUTPUT: a string with a coded message
+// FUNCTION: name letsMakeItUnreadable with a passed in parameter thing
+// PROCESS:
+// 1) create a function and pass in one parameter that will recognize array
+// 2) split the array and map through it, for to itirate and evaluate each value
+// 3) create a condition where value is replaced from letters to numbers
+// 4) join('') the result and recieve a readable string. BadaBimBadaBoom!!!!
+
+const letsMakeItUnreadable = (thing) => {
+    return thing.split('').map((value) => {
+        return value.replace("a", 4).replace("A", 4).replace("e", 3).replace("E", 3).replace("i",  1 ).replace("I", 1).replace("o",  0 ).replace("O", 0)
+    }).join('')
+} 
+
+
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+describe("letsMakeItUnreadable", () => {
+    
+    const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+    
+    const filterLetterA = "a"
+    // Expected output: ["Mango", "Apricot", "Peach"]
+    const filterLetterE = "e"
+    // Expected output: ["Cherry", "Blueberry", "Peach"]
+  
+    it("takes in an array of words and a single letter and returns an array of all the words containing that particular letter.", () => {
+      expect(filterIt(fruitArray, filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+      expect(filterIt(fruitArray, filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+  })
 
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
 
+// INPUT: an array of words, and a string with a letter
+// OUTPUT: and array of words in the string that correspond with a passed in letter
+// FUNCTION: name filterIt with 2 parameters array and letter
+// PROCESS:
+// 1) create a function that takes will be able to take in 2 arguments
+// 2) map through the first parameter
+// 3) create a conditional statement if value inclueds the 2nd parameter return it
+// 4) since this will return the entire array with true & unidentifed values we filter the outcome to return only boolean values. BadaBimBadaBoom!!!!
```

Comment:
The `.filter(Boolean)` is not quite returning only boolean values. By passing in Boolean as the argument in .filter, it is removing any falsy values, such as `undefined`, `false`, `NaN`, etc.